third_party/terraform/utils/common_polling.go

```diff
@@ -31,19 +32,75 @@ func SuccessPollResult() PollResult {
 	return nil
 }
 
-func PollingWaitTime(pollF PollReadFunc, checkResponse PollCheckResponseFunc, activity string, timeout time.Duration) error {
+func PollingWaitTime(pollF PollReadFunc, checkResponse PollCheckResponseFunc, activity string,
+	timeout time.Duration, targetOccurrences int) error {
 	log.Printf("[DEBUG] %s: Polling until expected state is read", activity)
-	return resource.Retry(timeout, func() *resource.RetryError {
+	log.Printf("[DEBUG] Target occurrences: %d", targetOccurrences)
+	return RetryWithTargetOccurrences(timeout, targetOccurrences, func() *resource.RetryError {
```

Comment:
It feels dangerous to me to replace this call with a significantly different code path when most of the calls to `PollingWaitTime` will have `targetOccurrences = 1`. Can we short circuit inside this method if `targetOccurrences = 1` and use the old `resource.Retry` logic?

This will *probably* not cause any issues but changing all of the existing calls to this method to involve locks and `resource.StateChangeConf` seems unnecessary